From 18af014f7c09aec79b68e8c404b18e4bc2ca7f59 Mon Sep 17 00:00:00 2001
From: Faith Ekstrand <faith.ekstrand@collabora.com>
Date: Thu, 19 Oct 2023 07:25:27 -0500
Subject: [PATCH 581/834] nvk: Re-arrange Vulkan 1.2 features to match the
 header

Also, conditional render ended up in the Vulkan 1.2 section which is
wrong.

Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25812>
---
 src/nouveau/vulkan/nvk_physical_device.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/nouveau/vulkan/nvk_physical_device.c b/src/nouveau/vulkan/nvk_physical_device.c
index fd271e1c27d..c7a9faa5e45 100644
--- a/src/nouveau/vulkan/nvk_physical_device.c
+++ b/src/nouveau/vulkan/nvk_physical_device.c
@@ -210,6 +210,7 @@ nvk_get_device_features(const struct nv_device_info *info,
 
       /* Vulkan 1.2 */
       .samplerMirrorClampToEdge = true,
+      .drawIndirectCount = info->cls_eng3d >= TURING_A,
       .shaderInputAttachmentArrayDynamicIndexing = true,
       .shaderUniformTexelBufferArrayDynamicIndexing = true,
       .shaderStorageTexelBufferArrayDynamicIndexing = true,
@@ -230,6 +231,7 @@ nvk_get_device_features(const struct nv_device_info *info,
       .descriptorBindingPartiallyBound = true,
       .descriptorBindingVariableDescriptorCount = true,
       .runtimeDescriptorArray = true,
+      .samplerFilterMinmax = info->cls_eng3d >= MAXWELL_B,
       .imagelessFramebuffer = true,
       .uniformBufferStandardLayout = true,
       .separateDepthStencilLayouts = true,
@@ -238,10 +240,6 @@ nvk_get_device_features(const struct nv_device_info *info,
       .bufferDeviceAddress = true,
       .bufferDeviceAddressCaptureReplay = false,
       .bufferDeviceAddressMultiDevice = false,
-      .drawIndirectCount = info->cls_eng3d >= TURING_A,
-      .samplerFilterMinmax = info->cls_eng3d >= MAXWELL_B,
-      .conditionalRendering = true,
-      .inheritedConditionalRendering = true,
 
       /* Vulkan 1.3 */
       .robustImageAccess = true,
@@ -276,6 +274,10 @@ nvk_get_device_features(const struct nv_device_info *info,
       /* VK_EXT_buffer_device_address */
       .bufferDeviceAddressCaptureReplayEXT = false,
 
+      /* VK_EXT_conditional_rendering */
+      .conditionalRendering = true,
+      .inheritedConditionalRendering = true,
+
       /* VK_EXT_custom_border_color */
       .customBorderColors = true,
       .customBorderColorWithoutFormat = true,
-- 
2.42.0

